src/osd/osd_types.cc

```diff
@@ -1943,6 +1945,8 @@ void object_stat_sum_t::decode(bufferlist::iterator& bl)
 #endif
   if (!decode_finish) {
     ::decode(num_bytes, bl);
+    if (struct_v >= 17)
+      ::decode(num_bytes_overlap, bl);
```

Comment:
this must go at the very end of the struct (or else you need feature-dependent encoding)